Px.Utils/PxFile/Data/PxFileStreamDataReader.cs

```diff
@@ -256,94 +245,74 @@ public void Dispose()
 
         #region Private methods
 
-        private void SetDataStartIfZero()
+        private void SetReaderPositionIfZero()
         {
-            if (_dataStart != 0) return;
+            if (_stream.Position != 0) return;
             string dataKeyword = _conf.Tokens.KeyWords.Data;
             long start = StreamUtilities.FindKeywordPosition(_stream, dataKeyword, _conf);
             if (start == -1)
             {
                 throw new ArgumentException($"Could not find data keyword '{dataKeyword}'");
             }
-            _dataStart = start + dataKeyword.Length + 1; // +1 to skip the '='
+            _stream.Position = start + dataKeyword.Length + 1; // +1 to skip the '='
         }
 
         private async Task SetReaderPositionIfZeroAsync(CancellationToken? cancellationToken = null)
         {
-            if (_dataStart != 0) return;
+            if (_stream.Position != 0) return;
             string dataKeyword = _conf.Tokens.KeyWords.Data;
             long start = await StreamUtilities.FindKeywordPositionAsync(_stream, dataKeyword, _conf, cancellationToken);
             if (start == -1)
             {
                 throw new ArgumentException($"Could not find data keyword '{dataKeyword}'");
             }
-            _dataStart = start + dataKeyword.Length + 1; // +1 to skip the '='
+            _stream.Position = start + dataKeyword.Length + 1; // +1 to skip the '='
         }
 
-        private void ReadItemsFromStream<T>(T[] buffer, int offset, int[] rows, int[] cols, Func<char[], int, T> readItem, CancellationToken? token = null)
+        private void ReadItemsFromStreamByCoordinate<T>(T[] buffer, int offset, DataIndexer indexer, Func<char[], int, T> readItem, CancellationToken? token = null)
         {
-            _stream.Position = _dataStart;
             int startingIndex = offset;
 
-            int colsLength = cols.Length;
-            int rowsLength = rows.Length;
-
             byte[] internalBuffer = new byte[INTERNAL_BUFFER_SIZE];
             char[] valueBuffer = new char[30]; // separate buffer is needed because a value can be split between two reads
-
-            int targetRowIndex = 0;
-            int targetColIndex = 0;
+            int valueBufferIndex = 0;
 
             long read;
-            int valueIndex = 0;
 
             do
             {
                 token?.ThrowIfCancellationRequested();
                 read = _stream.Read(internalBuffer, 0, INTERNAL_BUFFER_SIZE);
```

Comment:
Can we add a test where INTERNAL_BUFFER_SIZE would be smaller than test data set

Reply:
Changed it and added unit tests according to the feedback